File: ABC/251-300/260/c.py
n,x,y = map(int,input().split())
red = [0]*12
blue = [0]*12
red[0] = 1

for i in range(n-1):
    blue[i] = red[i]*x
    red[i+1] = red[i] + blue[i]
    blue[i+1] = y*blue[i]
print(red)
print(blue)
print(blue[n-2])

File: ABC/251-300/260/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<(n);i++)
using ll = long long;

int main(){
    ll n,x,y;
    cin >> n >> x >> y;
    vector<ll> red(12),blue(12);
    red[1] = 0; blue[1] = 1;
    for(int i=2;i<=n;i++){
        blue[i] = red[i-1] + blue[i-1]*y;
        red[i] = red[i-1] + blue[i]*x;
    }
    cout << red[n] << endl; 
    return 0;
}